source/extensions/filters/network/common/redis/redis_command_stats.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/stats/scope.h"
+#include "envoy/stats/timespan.h"
+
+#include "common/common/to_lower_table.h"
+#include "common/stats/symbol_table_impl.h"
+
+#include "extensions/filters/network/common/redis/codec.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace Common {
+namespace Redis {
+
+class RedisCommandStats {
+public:
+  RedisCommandStats(Stats::Scope&, const std::string& prefix, bool enabled);
+
+  // TODO: Make std::shared_ptr<RedisCommandStats> be RedisCommandStatsPtr
+  static std::shared_ptr<RedisCommandStats>
+  createRedisCommandStats(Stats::Scope& stats_scope, const std::string& prefix, bool enabled) {
+    auto redis_command_stats = std::make_shared<Common::Redis::RedisCommandStats>(
+        stats_scope, prefix,
+        enabled); // TODO: createScope uses make_unique, should I be doing that?
```

Comment:
You don't want to use make_unique here because we want to share the RedisCommandStats across multiple upstreams. So let's remove stats_scope from the arg list here and the member variables, and instead just pass and store the SymboLTable, which will be common across them. Make sense?

Reply:
Ok done